### ISSUE TITLE
Sample sentence required fields on migration

### ISSUE THREAD

**Issue by michael-wynne-wsu** [MukurtuCMS/Mukurtu-CMS#571]
- Ran a v3 to v4 migration. 
- A dictionary word included a word entry that had no sample sentence(s).
- When I tried to edit that word, I couldn't save it, the system was requiring a sample sentence (should not happen).
- When creating a new dictionary word on the site, it did not require a sample sentence (this is the correct behaviour).
- Additionally, the sample sentence field label is displayed even if there are no sample sentences entered (see attached).

![Screenshot 2024-08-21 at 2 28 56 PM](https://github.com/user-attachments/assets/7b4d125d-08c7-495a-9273-3896fa8c2216)
![Screenshot 2024-08-21 at 2 29 05 PM](https://github.com/user-attachments/assets/5dbd2158-0963-41d4-be8c-023485702166)

**Comment by michael-wynne-wsu** [comment]
@nick-deer

**Comment by nick-deer** [comment]
Additional context (copied from https://github.com/MukurtuCMS/Mukurtu-CMS/issues/192): 

@michael-wynne-wsu In v3, dictionary word sample sentences are implemented as a simple text field. However in v4, we made sample sentences their own paragraph complete with two fields: sample sentence text field and sample sentence recording field. The question is: do we want to make this change in v3 so that people can add sample sentence recordings as part of pre-migration steps, or do we let them manually set that up post-migration?

@nick-deer I'm fine not complicating the v3 sample sentence structure and if people want to do clean up post-migration they can. It's not a mandatory change like implementing protocols, and won't result in data loss etc.

HOWEVER, keep in mind that in v3, that simple text field accepts SCALD audio embeds, and those absolutely should be migrated (I'm not sure offhand if that's been done yet). If it's better to manage that in pre-migration, that would be a good reason to do it though.

**Comment by michael-wynne-wsu** [comment]
@nick-deer Just checking - activity here was cleanup, you didn't have any new questions, right?

**Comment by nick-deer** [comment]
@michael-wynne-wsu yeah that's right!

**Comment by michael-wynne-wsu** [comment]
[like]  Wynne, Michael reacted to your message:
________________________________

**Comment by m4olivei** [comment]
Hey @nick-deer and @michael-wynne-wsu . Just wanted to clarify, is the issue described here a bug up against the Draft PR? https://github.com/MukurtuCMS/Mukurtu-CMS/pull/1099

When I load up the given v3 site, first, it only has one Dictionary Word, and that one has 3 sample sentences. I also added one without any sample sentences. After running the migration, neither term had its sample sentences migrated. Neither term in v4 has any issues with edit/save.

**Comment by nick-deer** [comment]
> Hey [@nick-deer](https://github.com/nick-deer) and [@michael-wynne-wsu](https://github.com/michael-wynne-wsu) . Just wanted to clarify, is the issue described here a bug up against the Draft PR? [#1099](https://github.com/MukurtuCMS/Mukurtu-CMS/pull/1099)

No, the bug doesn't originate with the draft PR, that is just an attempt to migrate sample sentences. To date, there's been no other attempt to migrate them.

**Comment by michael-wynne-wsu** [comment]
After this meeting I can run a test migration and see exactly what's up, I haven't tested this in a bit honestly. 

- We have substantially changed the data structure of sample sentences, so need to make sure things migrate properly.

**Comment by michael-wynne-wsu** [comment]
Yes, looking at it again that's correct.  Nick confirmed that development of sample sentence migration was never completed (I misunderstood earlier).

Let me know if this better to show live (and Nick, note any corrections), but....

- V3
   - Sample sentences (text field)
   - Only within the word entry paragraph
   - Field contained either plain text or embedded scald atoms (typically an audio recording). I think it's very unlikely that anyone put BOTH text and and an embed in the same sentence.
   - No limit on number of sample sentences per word entry paragraph
- V4
   - Sample sentences (paragraph)
   - Contains a sample sentence text field AND a media asset field (limited to one audio file)
   - The "main" word, AND subsequent word entry paragraphs can have sample sentence paragraphs (unlimited)
- On migration, need to....
   - Ensure that the first v3 word entry and sample sentence field moves into the main word. Subsequent word entries will be word entries.
   - Convert each v3 sample sentence field to a v4 sample sentences paragraph.
   - Within that, scald embeds should be converted to media assets, while other text can be text sample sentences.

**Comment by michael-wynne-wsu** [comment]
@m4olivei Ah here are some!

- https://heritage.trtfn.com/dictionary-word/kag%C3%BA%E1%B8%B5x%CA%BCu
- https://heritage.trtfn.com/dictionary-word/aan

**Comment by m4olivei** [comment]
@nick-deer / @michael-wynne-wsu general migrations question for you, are we taking care to migrate revisions data over to v4?

**Comment by michael-wynne-wsu** [comment]
That's a great question that I'm not sure we've discussed in depth. Nick, do you have a sense of how we're handling that currently?

**Comment by michael-wynne-wsu** [comment]
Great, and I think that is acceptable to keep doing. Obviously it would not be a BAD idea, but given the major shift going on, and my sense that revisions are not really used much by users, as long as we include a notification of that, it's okay.

**Comment by m4olivei** [comment]
This is ready for review: https://github.com/MukurtuCMS/Mukurtu-CMS/pull/1170

**Comment by m4olivei** [comment]
Filed a follow up for handling mulitple Scald tokens if necessary: https://github.com/MukurtuCMS/Mukurtu-CMS/issues/1173
